### ISSUE TITLE
Find better place to store the generated xdocs

### ISSUE THREAD

**Issue by stoyanK7** [checkstyle/checkstyle#13512]
From https://github.com/checkstyle/checkstyle/pull/13499#discussion_r1284477777

> it bad that we put templates and same folder as xdoc files, better to not mix generated and not generated content, but this case is not simple
Lets discuss this in separate issue, please create issue on this.

We need to find a better place to store the generated xdocs. Having them in the same folder as the templates is not good.

**Comment by nrmancuso** [comment]
@stoyanK7 @romani any thoughts on a better location for these files? Maybe `/src/xdocs-templates/...`?

**Comment by stoyanK7** [comment]
@nrmancuso I am fine with this.

@romani I suppose this location is only until we get to the end (#13426)? After that, there would be no template and xdoc

**Comment by romani** [comment]
Yes, we will substitute xdoc files from modules with template files. But this will be at the end.

**Comment by MANISH-K-07** [comment]
@romani , @nrmancuso , @stoyanK7 , do we have a resolution here ?

I see that the generated xml are being stored at src/xdocs along with templates.
A little bit of clarity might help. So, after #13426 , we will only have _xml.template_ files (with macros) in the src/xdocs dir. As scope of the current issue, You want to change the target dir of our .xml files (generated xdocs) to a temporary folder ?

> any thoughts on a better location for these files? Maybe /src/xdocs-templates/...?

If this is the case, `/src/xdocs-templates/...` is a bit confusing since we are moving xdocs and not the templates.
And we could leave templates in permanent path src/xdocs , just configure xdocs to be generated at a new path something like `src/xdocs-generated/....`

Please do correct me in case my understanding is wrong :)

**Comment by romani** [comment]
lets keep it for now in same folder, we did not find any better way to do this. We might see what to do after completion of migration.
